tamr_client/dataset/record.py

```diff
@@ -143,3 +143,17 @@ def _delete_command(record: Dict, *, primary_key_name: str) -> Dict:
         The DELETE command in the proper format
     """
     return {"action": "DELETE", "recordId": record[primary_key_name]}
+
+
+def stream(session: Session, dataset: Dataset) -> Generator[Dict, None, None]:
```

Comment:
You should use `Iterator` type for generators that do not read from `yield` nor `return`. see https://docs.python.org/3/library/typing.html#typing.Generator

Also, you can use `types.JsonDict` instead of `Dict`

```suggestion
def stream(session: Session, dataset: Dataset) -> Iterator[JsonDict]:
```